### PR TITLE
fix(completion): prevent K_LUA from closing pum

### LINKED ISSUE #17024: Lua function keymap does not respect `pumvisible`
### Neovim version (nvim -v)

NVIM v0.7.0-dev+846-gbba679c43

### Vim (not Nvim) behaves the same?

No relevent

### Operating system/version

Void Linux

### Terminal name/version

st

### $TERM environment variable

st-256color

### Installation

From repo

### How to reproduce the issue

```
nvim --clean
:lua vim.keymap.set('i', '<Tab>', function() print(vim.fn.pumvisible()) end)
<C-X><C-F><Tab>
``

### Expected behavior

We should see a `1` appear here, indicating that, from the point of view of the function, the pum is visible.
It is worth noting that a mapping like the following works as expected:
```lua
function TAB()
  print(vim.fn.pumvisible())
end

vim.cmd [[imap <Tab> <cmd>lua TAB()<CR>]]

### Actual behavior

We see a bunch of zeros.

### PR DESCRIPTION
**Pull request by zeertzjq**
Fix #17024

### CODE REVIEW

**Comment by vigoux** [comment]
Maybe adding a test for this would be great ?
And I think we need experts' opinion on this: @bfredl

**Comment by zeertzjq** [comment]
I don't have much time this week, so I'm not sure when I'll have time to write tests for this.

**Comment by vigoux** [comment]
May I take that over then ?

**Comment by zeertzjq** [comment]
OK.

**Review by clason — APPROVED** [review]
LGTM!

**Comment by zeertzjq** [comment]
I made the test press the same key three times instead of once. Test fails on `master`:
```
[  ERROR   ] test/functional/api/keymap_spec.lua @ 823: nvim_set_keymap, nvim_del_keymap does not reset pum in lua mapping
test/helpers.lua:73: Expected objects to be the same.
Passed in:
(number) 0
Expected:
(number) 3
```

**Comment by vigoux** [comment]
We don't increment so it shouldn't be three either though

**Comment by zeertzjq** [comment]
I changed the test.

**Comment by vigoux** [comment]
I changed the test to test what you are asking for, and I can guarantee you that everything is fine.

File: test/functional/api/keymap_spec.lua [hunk 1]
```diff
@@ -819,6 +819,16 @@ describe('nvim_set_keymap, nvim_del_keymap', function()
 
     eq(99, exec_lua[[return SomeValue]])
   end)
+  
+  it('does not reset pum in lua mapping', function()
+    eq(0, exec_lua [[
+    was_visible = 0
```
Comment (bfredl) [inline test/functional/api/keymap_spec.lua:825]: indentation

**Review by bfredl — APPROVED** [review]

**Comment by vigoux** [comment]
Lol looks like you had time to do it in the end, sorry for bothering :wink:

**Pull request merged**